crates/bevy_ecs/src/schedule/config.rs

```diff
@@ -36,24 +36,68 @@ fn ambiguous_with(graph_info: &mut GraphInfo, set: InternedSystemSet) {
     }
 }
 
+/// Stores data to differentiate different Node types
+pub trait NodeType {
+    /// Used in `NodeConfig`
+    type Metadata;
+    /// Used in `NodeConfigs`
+    type GroupMetadata;
+
+    /// Initialize `NodeConfig` from `NodeType`
+    fn config(self) -> NodeConfig<Self>
```

Comment:
```suggestion
    fn into_config(self) -> NodeConfig<Self>
```
This change follows rust naming conventions for functions that take `self`.